File: Weekly/Week 1/keepbang/insertionSortList.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var insertionSortList = function(head) {
    if(head === null)
        return head;
    
    let result = new ListNode(null,new ListNode(head.val,null));
    
    head = head.next;
    
    while(head !== null){
        let tmp = new ListNode(head.val,null);
        let result_point = result;    
        
        while(result_point !== null){
            if(result_point.next === null){
                result_point.next = tmp;
                break;
            }
            if(result_point.next.val > tmp.val){
                tmp.next = result_point.next;
                result_point.next = tmp;
                break;
            }
            result_point = result_point.next;
        }
        
        
        
        head = head.next;
    }
    
    return result.next;
    
    
};

File: Weekly/Week 1/Jseastar/insertion_sort_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertionSortList(self, head: ListNode) -> ListNode:
        result = ListNode()
        curr = head
        
        while curr:
            prev_n = result
            next_n = prev_n.next
        
            while next_n:
                if curr.val < next_n.val:
                    break;
                prev_n = next_n
                next_n = next_n.next
                
            next_iter = curr.next
            
            curr.next = next_n
            prev_n.next = curr
            
            curr = next_iter
            
        return result.next
